hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1011,7 +1011,7 @@ public String startDeletePartitionCommit(HoodieTableMetaClient metaClient) {
    */
   @VisibleForTesting
   String startCommit(Option<String> providedInstantTime, String actionType, HoodieTableMetaClient metaClient) {
-    if (needsUpgrade(metaClient)) {
+    if (UpgradeDowngrade.needsUpgrade(metaClient, config, config.getWriteVersion())) {
```

Comment:
I feel like at some point we should refactor the need of passing config.writeVersion, since we are already passing config.